components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -394,17 +394,32 @@ public void updateOauthSecretKey(String consumerKey) throws IdentityOAuthAdminEx
 
     private void updateAppAndRevokeTokensAndAuthzCodes(String consumerKey, Properties properties) throws IdentityOAuthAdminException {
         TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
-
+        int countToken = 0;
         try {
-            Set<String> accessTokens = tokenMgtDAO.getActiveTokensForConsumerKey(consumerKey);
+            Set<AccessTokenDO> activeDetailedTokens = tokenMgtDAO.getActiveDetailedTokensForConsumerKey(consumerKey);
+            String[] accessTokens = new String[activeDetailedTokens.size()];
+
             if (OAuthServerConfiguration.getInstance().isCacheEnabled()) {
                 OAuthCache oauthCache = OAuthCache.getInstance();
-                for (String accessToken : accessTokens) {
-                    OAuthCacheKey cacheKey = new OAuthCacheKey(accessToken);
-                    oauthCache.clearCacheEntry(cacheKey);
+                for (AccessTokenDO detailToken : activeDetailedTokens) {
+                    String token = detailToken.getAccessToken();
+                    accessTokens[countToken] = token;
+                    countToken++;
+                    String scope = OAuth2Util.buildScopeString(detailToken.getScope());
+                    String authorizedUser = detailToken.getAuthzUser().toString();
+                    boolean isUsernameCaseSensitive = IdentityUtil.isUserStoreInUsernameCaseSensitive(authorizedUser);
+                    String cacheKeyString;
+                    if (isUsernameCaseSensitive) {
+                        cacheKeyString = consumerKey + ":" + authorizedUser + ":" + scope;
+                    } else {
+                        cacheKeyString = consumerKey + ":" + authorizedUser.toLowerCase() + ":" + scope;
```

Comment:
Shall we use string builder here ?

Reply:
It's fine to use string concatenation since the String concat internally uses StringBuilder[1]

[1] http://docs.oracle.com/javase/7/docs/api/java/lang/String.html

Reply:
done with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/472